lib/filter_nb.c

```diff
@@ -1285,13 +1302,79 @@ static int lib_prefix_list_entry_ipv4_prefix_length_lesser_or_equal_modify(
 	    prefix_list_length_validate(args) != NB_OK)
 		return NB_ERR_VALIDATION;
 
+	if (args->event != NB_EV_APPLY)
+		return NB_OK;
+
+	ple = nb_running_get_entry(args->dnode, NULL, true);
+
+	/* Start prefix entry update procedure. */
+	prefix_list_entry_update_start(ple);
+
+	ple->le = yang_dnode_get_uint8(args->dnode, NULL);
+
+	/* Finish prefix entry update procedure. */
+	prefix_list_entry_update_finish(ple);
+
+	return NB_OK;
+}
+
+static int lib_prefix_list_entry_ipv4_prefix_length_lesser_or_equal_destroy(
+	struct nb_cb_destroy_args *args)
+{
+	struct prefix_list_entry *ple;
+	int af_type;
+
+	if (args->event == NB_EV_VALIDATE) {
+		const struct lyd_node *plist_dnode =
+			yang_dnode_get_parent(args->dnode, "prefix-list");
+		af_type = yang_dnode_get_enum(plist_dnode, "./type");
+		if (af_type != YPLT_IPV4) {
```

Comment:
A common validate API, would help to check the type. You can call that API in
lib_prefix_list_entry_ipv4_prefix_modify
lib_prefix_list_entry_ipv4_prefix_length_greater_or_equal_destroy
lib_prefix_list_entry_ipv4_prefix_length_lesser_or_equal_destroy